google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/BigQueryImplTest.java

```diff
@@ -1470,6 +1471,41 @@ public void testListTableDataWithOptions() {
     verify(bigqueryRpcMock).listTableData(PROJECT, DATASET, TABLE, TABLE_DATA_LIST_OPTIONS);
   }
 
+  @Test
+  public void testListTableDataWithNextPage() {
+    doReturn(TABLE_DATA_PB)
+        .when(bigqueryRpcMock)
+        .listTableData(PROJECT, DATASET, TABLE, TABLE_DATA_LIST_OPTIONS);
+    bigquery = options.getService();
+    TableResult page =
+        bigquery.listTableData(
+            DATASET,
+            TABLE,
+            TABLE_DATA_LIST_PAGE_SIZE,
+            TABLE_DATA_LIST_PAGE_TOKEN,
+            TABLE_DATA_LIST_START_INDEX);
+    assertEquals(CURSOR, page.getNextPageToken());
+    verify(bigqueryRpcMock).listTableData(PROJECT, DATASET, TABLE, TABLE_DATA_LIST_OPTIONS);
+    assertArrayEquals(TABLE_DATA.toArray(), Iterables.toArray(page.getValues(), List.class));
+    Map<BigQueryRpc.Option, ?> SECOND_TABLE_DATA_LIST_OPTIONS =
+        ImmutableMap.of(BigQueryRpc.Option.PAGE_TOKEN, CURSOR, BigQueryRpc.Option.START_INDEX, 0L);
+    doReturn(
+            new TableDataList()
+                .setPageToken(null)
+                .setTotalRows(1L)
+                .setRows(
+                    ImmutableList.of(
+                        new TableRow().setF(ImmutableList.of(new TableCell().setV("Value3"))),
+                        new TableRow().setF(ImmutableList.of(new TableCell().setV("Value4"))))))
+        .when(bigqueryRpcMock)
+        .listTableData(PROJECT, DATASET, TABLE, SECOND_TABLE_DATA_LIST_OPTIONS);
+    if (page.hasNextPage()) {
+      page = page.getNextPage();
+    }
+    assertNull(page.getNextPageToken());
```

Comment:
`if (page.hasNextPage())`  => `assertTrue(page.hasNextPage())`
it's a test, you should know what is expected behavior.

Reply:
Done